packages/core/src/tracing/span.ts

```diff
@@ -227,6 +227,7 @@ export class Span implements SpanInterface {
    */
   public setHttpStatus(httpStatus: number): this {
     this.setTag('http.status_code', String(httpStatus));
+    this.setData('http.response.status_code', httpStatus);
```

Comment:
are we always sure this is going to be the status code of the response? What happens if it's the request? Shouldn't we be setting this on a case by case basis? I don't think we can get rid of the tag yet.

Reply:
I beleive http status code can be only on response.

Yes, I think the tag could be removed with next major.

Reply:
> What happens if it's the request?

@AbhiPrasad Does it really matter? The span is always gonna record the response's status code, no matter if this is instrumenting a client or a server http span.

Reply:
If the tag is removed, we break this feature https://github.com/getsentry/develop/issues/445
The FE bits have to adapt to use either the tag or the new `data` instead.
I guess this is only for transactions tho, not for spans.

Reply:
Yeah, I am not advocating for removing the tag. I am just saying it would be fine to set `http.response.status_code` in both cases - client span and server span.

Reply:
@lforst my reply was to https://github.com/getsentry/sentry-javascript/pull/8366#discussion_r1235390633